legacy/src/main/java/org/opensearch/sql/legacy/executor/cursor/CursorCloseExecutor.java

```diff
@@ -79,14 +83,25 @@ public String execute(Client client, Map<String, String> params) throws Exceptio
   }
 
   private String handleDefaultCursorCloseRequest(Client client, DefaultCursor cursor) {
-    String scrollId = cursor.getScrollId();
-    ClearScrollResponse clearScrollResponse =
-        client.prepareClearScroll().addScrollId(scrollId).get();
-    if (clearScrollResponse.isSucceeded()) {
-      return SUCCEEDED_TRUE;
+    if (LocalClusterState.state().getSettingValue(SQL_PAGINATION_API_SEARCH_AFTER)) {
+      String pitId = cursor.getPitId();
+      PointInTimeHandler pit = new PointInTimeHandlerImpl(client, pitId);
+      if (pit.delete()) {
+        return SUCCEEDED_TRUE;
+      } else {
+        Metrics.getInstance().getNumericalMetric(MetricName.FAILED_REQ_COUNT_SYS).increment();
+        return SUCCEEDED_FALSE;
+      }
```

Comment:
Does this makes sense, if we are never returning false from delete method. 91-93 would never get executed.

if you want to throw exception in delete method, catch exception here and emit metric.
If you want to return boolean, don't throw exception in delete method.

Reply:
I possible how scroll is executed below.

Reply:
Yes that's right! Updated the places using delete function to catch exception and log metrics similar to scroll.